sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -94,10 +94,18 @@ class InMemoryFileIndex(
     val files = listLeafFiles(rootPaths)
     cachedLeafFiles =
       new mutable.LinkedHashMap[Path, FileStatus]() ++= files.map(f => f.getPath -> f)
-    cachedLeafDirToChildrenFiles = files.toArray.groupBy(_.getPath.getParent)
+    cachedLeafDirToChildrenFiles = files.toArray.groupBy(
```

Comment:
What about this option? https://spark.apache.org/docs/latest/sql-data-sources-generic-options.html#recursive-file-lookup

Reply:
This action can't support read partitioned tables, and exception are thrown directly when handling partitioned table subdirectories.